// ngx-google-analytics/esm2020/lib/directives/ga-event-form-input.directive.mjs
import { Directive, Host, Optional, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./ga-event.directive";
export class GaEventFormInputDirective {
    constructor(gaEvent) {
        this.gaEvent = gaEvent;
        this.gaBind = 'focus';
    }
    set gaBind(bind) {
        if (this.gaEvent) {
            this.gaEvent.gaBind = bind;
        }
    }
}
GaEventFormInputDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: GaEventFormInputDirective, deps: [{ token: i1.GaEventDirective, host: true, optional: true }], target: i0.ɵɵFactoryTarget.Directive });
GaEventFormInputDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.0.2", type: GaEventFormInputDirective, selector: "input[gaEvent],\n             select[gaEvent],\n             textarea[gaEvent]", inputs: { gaBind: "gaBind" }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.2", ngImport: i0, type: GaEventFormInputDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: `input[gaEvent],
             select[gaEvent],
             textarea[gaEvent]`
                }]
        }], ctorParameters: function () { return [{ type: i1.GaEventDirective, decorators: [{
                    type: Host
                }, {
                    type: Optional
                }] }]; }, propDecorators: { gaBind: [{
                type: Input
            }] } });
